package/upgrade/lib.sh

```diff
@@ -184,3 +184,32 @@ detect_upgrade()
 
   UPGRADE_PREVIOUS_VERSION=$(echo "$upgrade_obj" | yq e .status.previousVersion -)
 }
+
+upgrade_addon()
+{
+  local name=$1
+  local namespace=$2
+  local version=$3
+
+  version=$(cat ./addons/${name}.yaml | yq e .spec.version)
+
+  cat > addon-patch.yaml <<EOF
+spec:
+  version: $version
+EOF
+
+  item_count=$(kubectl get addons.harvesterhci $name -n $namespace -o  jsonpath='{..name}')
+  if [ -z "$item_count" ]; then
```

Comment:
the CRD doesnt exist yet. The upgrade script will ship with the version packaging the addon CRD and onwards.

Reply:
I am not sure if we want to limit version upgrades from a specific version to another. Ideally with this, since the components are not core harvester you could skip versions if needed.